flake.nix

```diff
@@ -15,19 +15,24 @@
       url = "github:wavewave/replica/ghc-9.2";
       flake = false;
     };
-    hs-ogdf = {
-      url = "github:wavewave/hs-ogdf/master";
-      inputs.nixpkgs.follows = "nixpkgs";
-      inputs.flake-utils.follows = "flake-utils";
+    # This is needed to build the C++ library.
+    # TODO: replace the C++ library with ogdf in nixpkgs when the new version of
+    # ogdf (2022.02) is availabe in the upstream nixpkgs.
+    nixpkgs-ogdf.url = "github:wavewave/nixpkgs/wavewave/ogdf";
+    # Hackage version of OGDF
+    OGDF = {
+      url = "https://hackage.haskell.org/package/OGDF-1.0.0.0/OGDF-1.0.0.0.tar.gz";
+      flake = false;
     };
   };
-  outputs = { self, nixpkgs, flake-utils, concur, concur-replica, replica, hs-ogdf }:
+  outputs = inputs@{ self, nixpkgs, flake-utils, concur, concur-replica, replica, nixpkgs-ogdf, ... }:
```

Comment:
not necessary for this PR now, but other deps like concur, concur-replica .. had better be specified with explicit qualifier (i.e. inputs.concur).

Reply:
done.